File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)-1
        l, r = 0, n
        while l <= r:
            middle = (l + r) // 2
            if nums[middle] < target:
                l = middle+1
            elif nums[middle] > target:
                r = middle-1
            else:
                return middle
        return -1

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::set<int> mySet;

        for(int i = 0; i < nums.size(); i++){
           if(mySet.count(nums[i]) == 1){
                return true;
           }
           mySet.insert(nums[i]);
        }

        return false;
    }
};